lib/src/tileset.dart

```diff
@@ -29,17 +36,22 @@ class TilesetPreprocessor {
   /// Pre-processes a tileset to eliminate some expensive processing from
   /// the rendering stage.
   ///
-  /// returns a pre-processed tileset
-  Tileset preprocess(Tileset tileset) {
-    theme.layers.whereType<DefaultLayer>().forEach((themeLayer) {
-      themeLayer.selector.select(tileset).forEach((layer) {
-        themeLayer.selector.layerSelector
-            .features(layer.features)
-            .forEach((feature) {
-          feature.decodeGeometry();
-        });
-      });
-    });
-    return Tileset._preprocessed(tileset);
+  /// The [themeLayerFeatures] options controls whether for each theme layer,
+  /// the features it will render are pre-processed.
+  ///
+  /// Returns a pre-processed tileset.
+  Tileset preprocess(Tileset tileset, {bool themeLayerFeatures = true}) {
+    final themeLayerFeatureResolver = themeLayerFeatures
+        ? CachingThemeLayerFeatureResolver(tileset.themeLayerFeatureResolver)
```

Comment:
how about something like: `themeLayerFeatureResolver = tileset.themeLayerFeatureResolver is CachingThemeLayerFeatureResolver? tileset.themeLayerFeatureResolver: CachingThemeLayerFeatureResolver(tileset.themeLayerFeatureResolver)`